File: 633_Sum_of_Square_Numbers/sum_of_square.java
public class Solution {
    public boolean judgeSquareSum(int c) {
        int head = 0;
        int tail = (int)Math.sqrt(c);
        while (head <= tail) {
            int sum = head * head + tail * tail;
            if (sum == c) return true;
            else if (sum < c) head += 1;
            else tail -= 1;
        }
        return false;
    }
}

File: 633_Sum_of_Square_Numbers/sum_of_square.py
class Solution(object):
    def judgeSquareSum(self, c):
        """
        :type c: int
        :rtype: bool
        """
        tbl = set()
        for i in xrange(int(math.sqrt(c))+1):
            n = i * i
            tbl.add(n)
            if c - n in tbl:
                return True
        return False
